knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEngine.java

```diff
@@ -53,12 +54,17 @@ public ServiceEngine(Vertx vertx, TemplateEngineConfiguration serviceConfigurati
                     resp.subscribe(subscriber);
                     req.end();
                 });
-        return serviceResponse.flatMap(response ->
-                Observable.just(Buffer.buffer())
-                        .mergeWith(response.toObservable())
-                        .reduce(Buffer::appendBuffer))
-                .doOnNext(this::traceServiceCall)
-                .flatMap(buffer -> Observable.just(buffer.toJsonObject().getMap()));
+        return serviceResponse.flatMap(response -> Observable.just(Buffer.buffer())
+                .mergeWith(response.toObservable())
+                .reduce(Buffer::appendBuffer)
+                .flatMap(buffer -> Observable.just(buffer.toJsonObject().getMap()))
+                .map(results -> collectBuffers(results, response)));
+    }
+
+    private Map<String, Object> collectBuffers(Map<String, Object> results, HttpClientResponse response) {
```

Comment:
you missunderstood my intention. Take a look my previous comment once again. I was saying to extract this part:
```
Observable.just(Buffer.buffer()).mergeWith(response.toObservable()).reduce(Buffer::appendBuffer)
```
. 

Current this method is doing something else but not buffer reduction..